rapids-cmake/cpm/nvcomp.cmake

```diff
@@ -111,19 +111,37 @@ function(rapids_cpm_nvcomp)
     rapids_cpm_get_proprietary_binary_url(nvcomp ${version} nvcomp_url)
     if(nvcomp_url)
       rapids_cpm_download_proprietary_binary(nvcomp ${nvcomp_url})
-
-      # Unset CPM_DOWNLOAD_ALL if we have a proprietary binary enabled. This ensures we actually use
-      # the proprietary binary
-      unset(CPM_DOWNLOAD_ALL)
     endif()
 
-    # Record the nvcomp_DIR so that if USE_PROPRIETARY_BINARY is disabled we can safely clear the
-    # nvcomp_DIR value
     if(nvcomp_proprietary_binary)
-      set(nvcomp_proprietary_binary_dir "${nvcomp_ROOT}/lib/cmake/nvcomp")
+      # Set up the version of nvcomp we have downloaded to match the OS layout. that means ensuring
+      # we have a `lib64` directory on Fedora based machines
+      include("${rapids-cmake-dir}/cmake/install_lib_dir.cmake")
+      rapids_cmake_install_lib_dir(lib_dir)
+
+      # Replace ${_IMPORT_PREFIX}/lib/ with ${_IMPORT_PREFIX}/${lib_dir}/ in
+      # nvcomp-release-targets.cmake. Guarded in an EXISTS check so we only try to do this on the
+      # first configuration pass
+      if(NOT EXISTS "${nvcomp_ROOT}/${lib_dir}/cmake/nvcomp/nvcomp-targets-release.cmake")
+        file(READ "${nvcomp_ROOT}/lib/cmake/nvcomp/nvcomp-targets-release.cmake" FILE_CONTENTS)
+        string(REPLACE "\$\{_IMPORT_PREFIX\}/lib/" "\$\{_IMPORT_PREFIX\}/${lib_dir}/" FILE_CONTENTS
+                       ${FILE_CONTENTS})
+        file(WRITE "${nvcomp_ROOT}/lib/cmake/nvcomp/nvcomp-targets-release.cmake" ${FILE_CONTENTS})
+        file(RENAME "${nvcomp_ROOT}/lib/" "${nvcomp_ROOT}/${lib_dir}/")
+      endif()
+
+      # Record the nvcomp_DIR so that if USE_PROPRIETARY_BINARY is disabled we can safely clear the
+      # nvcomp_DIR value
+      set(nvcomp_proprietary_binary_dir "${nvcomp_ROOT}/${lib_dir}/cmake/nvcomp")
       cmake_path(NORMAL_PATH nvcomp_proprietary_binary_dir)
       set(rapids_cpm_nvcomp_proprietary_binary_dir "${nvcomp_proprietary_binary_dir}"
           CACHE INTERNAL "nvcomp proprietary location")
+
+      # Enforce that we need to find the local download of nvcomp and nothing else when we have a
+      # proprietary binary enabled.
+      list(APPEND CMAKE_PREFIX_PATH "${nvcomp_ROOT}/${lib_dir}/cmake/nvcomp")
+      unset(CPM_DOWNLOAD_ALL)
+      set(CPM_USE_LOCAL_PACKAGES ON)
```

Comment:
This isn't a cache variable, right? i.e. setting this here will only affect the `rapids_cpm_find` call below, and not any calls in consumers from before that? I'm assuming that there are no concerns about transitive finds in nvcomp itself (or at least, we can live with that if we are).

Reply:
That is all correct. Local variable so it will only effect `nvcomp` and no concerns about transitive finds